framework/src/main/java/org/checkerframework/common/value/util/Range.java

```diff
@@ -56,6 +57,26 @@
     /** The empty range. */
     public static final Range NOTHING = new Range();
 
+    /**
+     * Convenience construtor for often invoked pattern for Long collections
+     *
+     * @param collection the min and max value found in it will be used as from and to values in the
+     *     range
+     */
+    public Range(Collection<Long> collection) {
+        this(Collections.min(collection), Collections.max(collection));
```

Comment:
In #2905 you also mention the two passes through the collection as a concern.
Is it worth documenting this future improvement?

Reply:
Is it worth documenting for a tiny improvement or is it worth implementing and submitting with a ticket for a pull request ?  I am new to the group so I still don't know which has been teh preferred way.

We could spend pages on the analysis if we try to document it.
On the other hand, the code is veryshort... if passed a non-sorted  Collection<Number>...
//  traverses the list once, updates min and max only when needed.
//        long min=values.get(0).longValue();
//        long max=min;
//        for (Number value : values) {
//        	long current=value.longValue();
//        	if (min>current)
//        		min=current;
//        	if (max<current)
//        		max=current;
//        }
//        return new Range(min,max);
Total combined assignments to variables min and max = N 
One internal iterator setup. 

And local evaluations of binary comparison of two long variables, with one wasted iteration on the very first element.

Using Math.min() and max(), as in min=Math.min(min,current) are very legible but they do mean 2(N+1) assignments to variables min and max and function calls and returns for each element in the list.

Reply:
In my opinion, we don't need to document the potential improvement.  I suspect these lists will not be extremely long in general, and I suspect that if we find them to be a bottleneck, we will be able to figure out the improvement without a hint in a comment.

Reply:
One can not declare both convenience signatures to be Collection<Long> and Collection<Integer> because of generic type erasure.  Type is discarded on generics so the two signatures end up the same Range(Collection<?>)  and therefore the compiler does not allow them to co-exist.

Reply:
The build shows the problem: 
/__w/1/s/framework/src/main/java/org/checkerframework/common/value/util/Range.java:76: error: name clash: Range(Collection<Integer>) and Range(Collection<Long>) have the same erasure
    public Range(Collection<Integer> collection) {
           ^